src/main/kotlin/org/wycliffeassociates/sourceaudio/common/data/model/FileUploadModel.kt

```diff
@@ -23,9 +24,20 @@ data class FileUploadModel(
     @Throws(IllegalArgumentException::class)
     private fun validate() {
         if (languageCode.isBlank()) { throw IllegalArgumentException("Language Code is empty") }
-        if (resourceType.isBlank()) { throw IllegalArgumentException("Resource Type is empty") }
+        if (resourceType.isBlank()) { throw IllegalArgumentException("Dublin Core ID is empty") }
         if (grouping.isBlank()) { throw IllegalArgumentException("Group is empty") }
+        if (chapter.isNotEmpty()) {
+            if (projectId.isBlank()) {
+                throw IllegalArgumentException("BookID is not specified")
+            }
+            try {
+                chapter.toInt()
+            } catch (e: Exception) {
+                throw IllegalArgumentException("Chapter is invalid")
+            }
+        }
```

Comment:
I would suggest moving this to line 42 or so. Put it below all of the one-line if statements but put it above the validateExtensions function call. I think that would be more readable.